practices/managing-node-projects.md

```diff
@@ -18,54 +23,72 @@
+This is the method we recommend for installing dependencies when authoring node _applications_.
+
+### How to specify the version of node to use
+
+There are two main ways of doing this, and you're encouraged to do both.
+
+1. The [`engines`](https://docs.npmjs.com/files/package.json#engines) field in `package.json`:
+     * `engines` is important if authoring libraries. Let's assume we're using a different version of node to that specified in `engines` field for package `foo`.
+     When running `npm install` to install `foo`'s dependencies, `npm` will warn about the problem.
+     When `npm install`ing a package _which depends on_ `foo`, `npm` will warn.
+     * Some deployment environments also respect it.
+     * It makes compatibility requirements explicit to developers working on your application.
+2. Using an `.nvmrc` file:
+     * An `.nvmrc` file is a configuration file for `nvm` ([Node Version Manager](https://github.com/creationix/nvm)).
+     Use of `nvm` is strongly encouraged as an application author, as it enables you to specify to other developers what versions of node they should be using when running your app.
+     * Your application should include an [`.nvmrc` file](https://github.com/creationix/nvm#nvmrc) in the root directory of the project to specify which version(s) of node are compatible.
+     * It also is required for proper `npm ci` usage, as explained above.
+     * It also makes life much easier when working on multiple node projects locally that require different versions of node.
+
+#### `nvm` usage
+
+When you check out a project, you should first run `nvm use` before `npm ci` to ensure you're running on the same version of node the project expects. Additionally [Travis respects `.nvmrc` files](https://docs.travis-ci.com/user/languages/javascript-with-nodejs/#specifying-nodejs-versions-using-nvmrc), so using one will simplify your Travis configuration and help other developers avoid mistakes.
 
 #### Automatically running `nvm use`
 
-To avoid accidentally forgetting to run `nvm use` before working on a node project, there are useful shell extensions which will do it for you
-([zsh](https://github.com/creationix/nvm#calling-nvm-use-automatically-in-a-directory-with-a-nvmrc-file),
-[bash](https://stackoverflow.com/questions/23556330/run-nvm-use-automatically-every-time-theres-a-nvmrc-file-on-the-directory)),
-which are well worth installing!
+As mentioned above, it is important to run `nvm use` before `npm ci`. But this is a thing that's easy to forget!
+
+To avoid forgetting, there are useful shell extensions ([zsh](https://github.com/creationix/nvm#calling-nvm-use-automatically-in-a-directory-with-a-nvmrc-file),
```

Comment:
would it be useful to suggest that a `preinstall` can be used to run `nvm use` before anything is installed? it wouldn't help with changing between projects, but it would help when installing modules for the first time and could be a useful defensive alternative that a developer could implement

Reply:
Hmm, interesting point, thanks 🙂

I do get your argument, but in _this_ case I think it's probably better to crash early with errors on install TBH, to make developers aware of the issue. Anecdata but, I don't think I've ever used `preinstall` so this bit of magic might come as a bit of a surprise?

And yes as you say, it won't help with switching between projects (or the global node & npm versions being hiked). So am inclined to keep it simple...

Reply:
Yep, that seems a fair point. Better to keep things simple